backend/consensus/base.py

```diff
@@ -332,30 +409,31 @@ async def exec_transaction(
 
         # Update contract state if it is an existing contract
         else:
-            contract_snapshot.update_contract_state(leader_receipt.contract_state)
-
-        ConsensusAlgorithm.dispatch_transaction_status_update(
-            transactions_processor,
-            transaction.hash,
-            TransactionStatus.FINALIZED,
-            msg_handler,
-        )
+            contract_snapshot.update_contract_state(leader_receipt["contract_state"])
 
         # Finalize transaction
+        consensus_data["final"] = True
```

Comment:
It seems that this property is not used and it could be removed.

Reply:
Removed in eb5ea52a2d4433e9b5e2441d05bf6f027712d486